### ISSUE TITLE
BaseFilterSet not using multiple choice filters for CharFields with choices

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#3214]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.0a1
* Python version: 3.10
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. from nbshell inspect the filter for `Interface.mode`:
```py
from nautobot.dcim.filters import InterfaceFilterSet
InterfaceFilterSet.get_filters().get("mode")
# returns <django_filters.filters.ChoiceFilter object at 0x7ff8f8a57e80>
```

### Expected Behavior
This should be a `MultipleChoiceFilter`

### Observed Behavior
The filter is being set to `ChoiceFilter`

**Comment by bryanculver** [comment]
In the same family as #1889 and #1483

**Comment by bryanculver** [comment]
Issues like cause PRs like this: https://github.com/nautobot/nautobot/pull/2754#issuecomment-1311199557 when trying to adhere to standards defined here: https://github.com/nautobot/nautobot/issues/1606

**Comment by bryanculver** [comment]
Closed via #3342
